src/vs/base/common/arrays.ts

```diff
@@ -612,3 +612,78 @@ export function mapFind<T, R>(array: Iterable<T>, mapFn: (value: T) => R | undef
 
 	return undefined;
 }
+
+/**
+ * An alternative for Array.push(...items) method, Use this if you need to push large number of items to the array.
```

Comment:
Would suggest just putting these in our own arrays.ts class unless you're planning on trying to get VS Code to take it

Reply:
I need to use the splice method in a vscode file, here is the vscode PR https://github.com/microsoft/vscode/pull/119929